src/SignalR/common/Http.Connections/src/ConnectionEndpointRouteBuilderExtensions.cs

```diff
@@ -0,0 +1,151 @@
+using Microsoft.AspNetCore.Http.Connections;
+using Microsoft.AspNetCore.Http.Connections.Internal;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace Microsoft.AspNetCore.Routing
+{
+    public static class ConnectionEndpointRouteBuilderExtensions
+    {
+        /// <summary>
+        /// Maps incoming requests with the specified path to the provided connection pipeline.
+        /// </summary>
+        /// <param name="builder">The <see cref="IEndpointRouteBuilder"/> to add the route to.</param>
+        /// <param name="pattern">The route pattern.</param>
+        /// <param name="configure">A callback to configure the connection.</param>
+        /// <returns>An <see cref="IEndpointConventionBuilder"/> for endpoints associated with the connections.</returns>
+        public static IEndpointConventionBuilder MapConnections(this IEndpointRouteBuilder builder, string pattern, Action<IConnectionBuilder> configure) =>
+            builder.MapConnections(pattern, new HttpConnectionDispatcherOptions(), configure);
+
+        /// <summary>
+        /// Maps incoming requests with the specified path to the provided connection pipeline.
+        /// </summary>
+        /// <typeparam name="TConnectionHandler">The <see cref="ConnectionHandler"/> type.</typeparam>
+        /// <param name="builder">The <see cref="IEndpointRouteBuilder"/> to add the route to.</param>
+        /// <param name="pattern">The route pattern.</param>
+        /// <returns>An <see cref="IEndpointConventionBuilder"/> for endpoints associated with the connections.</returns>
+        public static IEndpointConventionBuilder MapConnectionHandler<TConnectionHandler>(this IEndpointRouteBuilder builder, string pattern) where TConnectionHandler : ConnectionHandler
+        {
+            return builder.MapConnectionHandler<TConnectionHandler>(pattern, configureOptions: null);
+        }
+
+        /// <summary>
+        /// Maps incoming requests with the specified path to the provided connection pipeline.
+        /// </summary>
+        /// <typeparam name="TConnectionHandler">The <see cref="ConnectionHandler"/> type.</typeparam>
+        /// <param name="builder">The <see cref="IEndpointRouteBuilder"/> to add the route to.</param>
+        /// <param name="pattern">The route pattern.</param>
+        /// <param name="configureOptions">A callback to configure dispatcher options.</param>
+        /// <returns>An <see cref="IEndpointConventionBuilder"/> for endpoints associated with the connections.</returns>
+        public static IEndpointConventionBuilder MapConnectionHandler<TConnectionHandler>(this IEndpointRouteBuilder builder, string pattern, Action<HttpConnectionDispatcherOptions> configureOptions) where TConnectionHandler : ConnectionHandler
+        {
+            var options = new HttpConnectionDispatcherOptions();
+            // REVIEW: WE should consider removing this and instead just relying on the
+            // AuthorizationMiddleware
```

Comment:
What part are you sayign we should remove? Are you talking about `options.AuthorizationData`?

I'd say that what's curently here is right since without knowing more I woulnd't suggest breaking an existing API.